tsl/src/compression/algorithms/array.h

```diff
@@ -14,9 +14,12 @@
 
 #include <postgres.h>
 #include <fmgr.h>
+#include <utils/memutils.h>
 
 #include "compression/compression.h"
 
+#define MAX_ARRAY_COMPRESSOR_SIZE_BYTES MaxAllocSize
```

Comment:
Let's make it a GUC right away? I think it might be useful when you compress very big values and want the batches of manageable size, especially if we increase the batch row count.

Reply:
IDK about usefulness of this. Seems very hard to use predictably. I'd leave it as a hard limit since this is the max we can support. Can always revisit at a later point.